model-engine/model_engine_server/api/v2/chat_completion.py

```diff
@@ -0,0 +1,258 @@
+import traceback
+from datetime import datetime
+
+import pytz
+from fastapi import APIRouter, BackgroundTasks, Depends, HTTPException
+from model_engine_server.api.dependencies import (
+    ExternalInterfaces,
+    get_external_interfaces_read_only,
+    verify_authentication,
+)
+from model_engine_server.common.config import hmi_config
+from model_engine_server.common.dtos.llms import (
+    ChatCompletionV2ErrorChunk,
+    ChatCompletionV2Request,
+    ChatCompletionV2Response,
+    ChatCompletionV2ResponseItem,
+    StreamError,
+    StreamErrorContent,
+    TokenUsage,
+)
+from model_engine_server.core.auth.authentication_repository import User
+from model_engine_server.core.loggers import (
+    LoggerTagKey,
+    LoggerTagManager,
+    logger_name,
+    make_logger,
+)
+from model_engine_server.core.utils.timer import timer
+from model_engine_server.domain.exceptions import (
+    EndpointUnsupportedInferenceTypeException,
+    EndpointUnsupportedRequestException,
+    InvalidRequestException,
+    ObjectHasInvalidValueException,
+    ObjectNotAuthorizedException,
+    ObjectNotFoundException,
+    UpstreamServiceError,
+)
+from model_engine_server.domain.gateways.monitoring_metrics_gateway import MetricMetadata
+from model_engine_server.domain.use_cases.llm_model_endpoint_use_cases import (
+    ChatCompletionStreamV2UseCase,
+    ChatCompletionSyncV2UseCase,
+)
+from sse_starlette import EventSourceResponse
+
+from .common import get_metric_metadata, record_route_call
+
+logger = make_logger(logger_name())
+
+chat_router_v2 = APIRouter(dependencies=[Depends(record_route_call)])
+
+
+def handle_streaming_exception(
+    e: Exception,
+    code: int,
+    message: str,
+):
+    tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+    request_id = LoggerTagManager.get(LoggerTagKey.REQUEST_ID)
+    timestamp = datetime.now(pytz.timezone("US/Pacific")).strftime("%Y-%m-%d %H:%M:%S %Z")
```

Comment:
should we make this UTC? otherwise it could get confusing if stuff isn't deployed on US west coast

Reply:
true, though i just made this consistent wit the existing completion APIs